plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -67,16 +87,10 @@
 import org.apache.maven.shared.filtering.MavenFileFilter;
 import org.apache.maven.shared.filtering.MavenFilteringException;
 
-import javax.validation.ConstraintViolationException;
-import java.io.File;
-import java.io.FileFilter;
-import java.io.IOException;
-import java.util.*;
-
-import static io.fabric8.maven.core.util.Constants.*;
+import static io.fabric8.maven.core.util.Constants.RESOURCE_APP_CATALOG_ANNOTATION;
 import static io.fabric8.maven.core.util.KubernetesResourceUtil.isAppCatalogResource;
+import static io.fabric8.maven.core.util.KubernetesResourceUtil.loadResources;
 import static io.fabric8.maven.plugin.mojo.build.ApplyMojo.DEFAULT_OPENSHIFT_MANIFEST;
-import static io.fabric8.maven.plugin.mojo.build.ApplyMojo.loadResources;
```

Comment:
awesome that get rid of this coupling. Let's think also where to move the DEFAULT_OPENSHIFT_MANIFEST constant.

Have I already mentioned that I'm not a big fan of [classes only for constants](http://stackoverflow.com/a/66228/207604) ? (always so negative, those Germans ;-) It might be harder to think a bit for a concrete class where to put the constant on (ideally its used only within a single class anyway), but it reduces coupling and increases [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science))

But, well, probably belongs to the category 'matter of taste'. I only hate when unrelated constants are collected in a `utils.Constants` class because its _technically_ easy since this class is reachable from every (but will hurt for maintenance or when moving things around). Don't know whether this the case here, though.

(as a general note, and because I'm already at it ;-), the same applies for "util" classes which are also accessible from everywhere. It's so tempting to stuff all things into utility classes because so you don't have to think about dependencies.)

rant over ;-)